tests/phpunit/tests/test-rest-languages.php

```diff
@@ -0,0 +1,243 @@
+			'direction' => 'rtl',
+			'flag'      => 'be',
+			'order'     => 22,
+		);
+		$request = new WP_REST_Request( 'POST', '/pll/v2/languages' );
+		$request->set_param( 'locale', 'fr_FR' ); // Required.
+		$request->set_param( 'w3c', 'foo' ); // Should not be set.
+		$request->set_param( 'set_default_cat', false ); // Do not create the default category for this language.
+
+		foreach ( $values as $name => $value ) {
+			$request->set_param( $name, $value );
+		}
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+
+		// Check the values match.
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+
+		foreach ( $values as $name => $value ) {
+			$this->assertSame( $value, $data[ $name ] );
+		}
+
+		$this->assertNotSame( 'foo', $data['w3c'] );
+
+		// Check the default category is not created.
+		$def_cat_id_fr = $this->pll_env->model->term->get( $def_cat_id, $data['code'] );
+		$this->assertSame( 0, $def_cat_id_fr );
+	}
+
+	public function test_get_language() {
+		self::create_language( 'fr_FR' );
+		self::create_language( 'en_US' );
+
+		$request = new WP_REST_Request( 'GET', '/pll/v2/languages/fr' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+		$this->assertArrayHasKey( 'code', $data );
+		$this->assertSame( 'fr', $data['code'] );
+	}
+
+	public function test_update_language() {
+		wp_set_current_user( self::$administrator );
+		self::create_language( 'fr_FR' );
+
+		$values = array(
+			'locale'    => 'fr_BE',
+			'name'      => 'François',
+			'new_code'  => 'fra',
+			'direction' => 'rtl',
+			'flag'      => 'be',
+			'order'     => 22,
+		);
+		$request = new WP_REST_Request( 'PATCH', '/pll/v2/languages/fr' );
+
+		foreach ( $values as $name => $value ) {
+			$request->set_param( $name, $value );
+		}
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+
+		foreach ( $values as $name => $value ) {
+			$name = 'new_code' === $name ? 'code' : $name;
+			$this->assertSame( $value, $data[ $name ] );
+		}
+	}
+
+	public function test_delete_language() {
+		wp_set_current_user( self::$administrator );
+		self::create_language( 'fr_FR' );
+
+		$request  = new WP_REST_Request( 'DELETE', '/pll/v2/languages/fr' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+		$this->assertArrayHasKey( 'deleted', $data );
+		$this->assertTrue( $data['deleted'] );
+		$this->assertArrayHasKey( 'previous', $data );
+		$this->assertIsArray( $data['previous'] );
+
+		$this->assertFalse( $this->pll_env->model->get_language( 'fr_FR' ) );
+	}
+
+	public function test_permissions() {
```

Comment:
Maybe we should have a test for `403` as well, what do you think?

Reply:
Done in [418d7a0](https://github.com/polylang/polylang/pull/1505/commits/418d7a081fb0670b016e85b4ddf7ca89d4eee83e).